### ISSUE TITLE
pylint doesn't recognize asyncio.subprocess

### ISSUE THREAD

**Issue by thanatos** [PyCQA/pylint#1469]
### Steps to reproduce
1. Using this test file, `example.py`:
   ```python
   from asyncio import subprocess

   subprocess.create_subprocess_exec('echo', 'hi')
   ```

2. Run `pylint example.py`

### Current behavior

The following output:

```
» pylint example.py
No config file found, using default configuration
************* Module example
C:  1, 0: Missing module docstring (missing-docstring)
E:  3, 0: Module 'subprocess' has no 'create_subprocess_exec' member (no-member)

[ more irrelevant output follows ]
```

### Expected behavior
The following warning is emitted:
```
E:  3, 0: Module 'subprocess' has no 'create_subprocess_exec' member (no-member)
```

But the "`subprocess`" (`asyncio.subprocess`) module _does_ have that member. It appears to be getting confused with the non-async `subprocess` module: if I do `subprocess.Popen(…)`, that _doesn't_ emit a warning, and should. The import `from asyncio import subprocess` isn't exactly trying to throw Pylint off the scent here.

### pylint --version output
```
» pylint --version
No config file found, using default configuration
pylint 1.6.5,
astroid 1.4.9
Python 3.6.1 (default, Mar 23 2017, 16:49:06)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

**Comment by rogalski** [comment]
I did a little troubleshooting but wasn't able to fix it yet.

Seems like a bug is in `ImportFrom._infer`. Goven following tescases, first one passes while second one fails.
```python
class TestClass(unittest.TestCase):
    def test_import_aliased(self):
        code = '''
        import asyncio.subprocess as subprocess
        subprocess'''
        module_name_node = extract_node(code)
        self.assertEqual('asyncio.subprocess', next(module_name_node.infer()).qname())

    def test_import_from_aliased(self):
        code = '''
        from asyncio import subprocess
        subprocess'''
        module_name_node = extract_node(code)
        self.assertEqual('asyncio.subprocess', next(module_name_node.infer()).qname())
```

[`bases._infer_stmts`](https://github.com/PyCQA/astroid/blob/master/astroid/inference.py#L227) is unable to resolve `subprocess` import to subpackage given inputs passed to it.

**Comment by JulienPalard** [comment]
For the record I can still reproduce it with
```
pylint 2.1.1
astroid 2.0.4
Python 3.6.6 (default, Jun 27 2018, 14:44:17) 
[GCC 8.1.0]
```

**Comment by PCManticore** [comment]
Thanks @JulienPalard This hasn't been fixed yet but we'll get to it.

**Comment by Wiezzel** [comment]
Well... it's 2021, I'm using pylint 2.9.3 with Python 3.8.0 and still getting this issue.

**Comment by Pierre-Sassoulas** [comment]
The maintainer you're responding to burnt out and is now in another field. But you can be the change you want to see in the world and open a merge request that fix the issue 😉

**Comment by cdce8p** [comment]
`asyncio` is a somewhat strange module. As I understand it from the docs, it's designed to always be imported as
```py
import asyncio
```

Thus, the correct usage in this case would be this. Which works in `pylint`.
```py
import asyncio

asyncio.create_subprocess_exec('echo', 'hi')
```
https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.create_subprocess_exec

--
Fyi: From the examples sections on how to use symbols from `asyncio.subprocess`
```py
import asyncio

asyncio.subprocess.PIPE
```
https://docs.python.org/3/library/asyncio-subprocess.html#examples

**Comment by firesock** [comment]
Following up on the last comment, trying to use `asyncio.subprocess.Process` appears as `E1101: Module 'subprocess' has no 'Process' member (no-member)` even without using a `from` anywhere, so that doesn't appear to be a complete solution yet.

Tested with:
```
pylint 2.11.1
astroid 2.8.4
Python 3.8.11 (default, Jul 22 2021, 15:41:12) 
[GCC 8.3.0]
```

**Comment by BoniLindsley** [comment]
I think I might have found an [MWE](https://github.com/BoniLindsley/pylint-issue-1469) for this.

[Typestub for `asyncio`](https://github.com/python/typeshed/blob/9c212cdf5c1573f71d4fa55f02cfd058e2a4a6e3/stdlib/asyncio/__init__.pyi#L13) uses a wildcard import of its submodule `subprocess.pyi`. But this submodule imports the standard library `subprocess`. Depending on how `asyncio.subprocess` is looked up, it might resolve to different nodes.

```py
from asyncio.subprocess import Process  # Imports directly from `asyncio/subprocess.pyi

import asyncio
asyncio.subprocess  # This is actually the standard library according to typestub.
asyncio.subprocess.Process  # Leading to `no-member` error.
```

If I understand this correctly, Pylint has the right behaviour, and typeshed might need updating.

**Comment by hannesweisbach** [comment]
I have the same issue with

```bash
$ pylint --version
pylint 3.1.0
astroid 3.1.0
Python 3.11.6 (main, Apr 10 2024, 17:26:07) [GCC 13.2.0]
```

and stumbled upon this thread.

`from asyncio.subprocess import Process`, as mentioned by @BoniLindsley worked for me as well.

From what @cdce8p mentioned on the usage of `asyncio`, I tried the annotation `asyncio.Process`, which worked for me:
```py
import asyncio

process: asyncio.Process
```

 I'm wondering if this is what should be used?

**Comment by BoniLindsley** [comment]
Does `asyncio.Process` cause a runtime error? IIRC, this is defined only in the typestub due to a wildcard import.

**Comment by hannesweisbach** [comment]
It works as annotation. I don't think you're supposed to instantiate `asyncio.Process` directly and thus haven't tried it.

**Comment by hannesweisbach** [comment]
Hm, now it failed in the CI. I'm not sure what changed or if I made a mistake previously. So I guess you're right after all.
